c10/CMakeLists.txt

```diff
@@ -12,120 +12,127 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 # protobuf header files, because protobuf header files will transitively force
 # one to link against a specific protobuf version.
 
-# ---[ Configure macro file.
-set(C10_USE_GFLAGS ${USE_GFLAGS}) # used in cmake_macros.h.in
-set(C10_USE_GLOG ${USE_GLOG}) # used in cmake_macros.h.in
-set(C10_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS}) # used in cmake_macros.h.in
-set(C10_USE_NUMA ${USE_NUMA})
-set(C10_USE_MSVC_STATIC_RUNTIME ${CAFFE2_USE_MSVC_STATIC_RUNTIME})
-set(C10_USE_ROCM_KERNEL_ASSERT ${USE_ROCM_KERNEL_ASSERT})
-configure_file(
-    ${CMAKE_CURRENT_LIST_DIR}/macros/cmake_macros.h.in
-    ${CMAKE_BINARY_DIR}/c10/macros/cmake_macros.h)
-
-# Note: if you want to add ANY dependency to the c10 library, make sure you
-# check with the core PyTorch developers as the dependency will be
-# transitively passed on to all libraries dependent on PyTorch.
-file(GLOB C10_SRCS
-        *.cpp
-        core/*.cpp
-        core/impl/*.cpp
-        mobile/*.cpp
-        macros/*.cpp
-        util/*.cpp
-      )
-file(GLOB C10_HEADERS
-        *.h
-        core/*.h
-        core/impl/*.h
-        mobile/*.h
-        macros/*.h
-        util/*.h
-      )
-add_library(c10 ${C10_SRCS} ${C10_HEADERS})
-target_compile_options_if_supported(c10 "-Wdeprecated")
-if(HAVE_SOVERSION)
-  set_target_properties(c10 PROPERTIES
-      VERSION ${TORCH_VERSION} SOVERSION ${TORCH_SOVERSION})
-endif()
-# If building shared library, set dllimport/dllexport proper.
-target_compile_options(c10 PRIVATE "-DC10_BUILD_MAIN_LIB")
-# Enable hidden visibility if compiler supports it.
-if(${COMPILER_SUPPORTS_HIDDEN_VISIBILITY})
-  target_compile_options(c10 PRIVATE "-fvisibility=hidden")
+if(BUILD_LIBTORCHLESS)
+  find_library(C10_LIB c10 PATHS $ENV{LIBTORCH_LIB_PATH} NO_DEFAULT_PATH)
 endif()
```

Comment:
```else()
  set(C10_LIB c10)
endif()
```

This will be cleaner if we use C10_LIB everywhere in this case, not mix of c10 and C10_LIB